File: Java-Core/src/main/java/prompto/javascript/JavaScriptMemberExpression.java
package prompto.javascript;

import prompto.utils.CodeWriter;

public class JavaScriptMemberExpression extends JavaScriptSelectorExpression {

	String name;
	
	public JavaScriptMemberExpression(String name) {
		this.name = name;
	}

	@Override
	public String toString() {
		return parent.toString() + "." + name;
	}

	@Override
	public void toDialect(CodeWriter writer) {
		parent.toDialect(writer);
		writer.append('.');
		writer.append(name);
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/javascript/JavaScriptMemberExpression.js
var JavaScriptSelectorExpression = require("./JavaScriptSelectorExpression").JavaScriptSelectorExpression;

function JavaScriptMemberExpression(name) {
	JavaScriptSelectorExpression.call(this);
	this.name = name;
	return this;
}

JavaScriptMemberExpression.prototype = Object.create(JavaScriptSelectorExpression.prototype);
JavaScriptMemberExpression.prototype.constructor = JavaScriptMemberExpression;

JavaScriptMemberExpression.prototype.toString = function() {
	return this.parent.toString() + "." + this.name;
};

JavaScriptMemberExpression.prototype.interpret = function(context) {
    var o = this.parent.interpret(context);
    if(o!=null) {
        return this.interpret_field(o);
    } else {
        return null;
    }
};

JavaScriptMemberExpression.prototype.toDialect = function(writer) {
    if (this.parent != null) {
        this.parent.toDialect(writer);
        writer.append('.');
    }
    writer.append(this.name);
};

JavaScriptMemberExpression.prototype.interpret_field = function(o) {
    return o[this.name];
};

exports.JavaScriptMemberExpression = JavaScriptMemberExpression;